installutils/helminstall/client.go

```diff
@@ -0,0 +1,205 @@
+type HelmInstall interface {
+	Run(chrt *chart.Chart, vals map[string]interface{}) (*release.Release, error)
+}
+
+// an interface around Helm's action.Uninstall struct
+type HelmUninstall interface {
+	Run(name string) (*release.UninstallReleaseResponse, error)
+}
+
+var _ HelmInstall = &action.Install{}
+var _ HelmUninstall = &action.Uninstall{}
+
+// interface around needed afero functions
+type TempFile interface {
+	NewTempFile(fs afero.Fs, dir, prefix string) (f afero.File, err error)
+	WriteFile(fs afero.Fs, filename string, data []byte, perm os.FileMode) error
+}
+
+type tempFile struct{}
+
+func NewTempFile() TempFile {
+	return &tempFile{}
+}
+
+func (a *tempFile) NewTempFile(fs afero.Fs, dir, prefix string) (f afero.File, err error) {
+	return afero.TempFile(fs, dir, prefix)
+}
+
+func (a *tempFile) WriteFile(fs afero.Fs, filename string, data []byte, perm os.FileMode) error {
+	return afero.WriteFile(fs, filename, data, perm)
+}
+
+// a HelmClient that talks to the kube api server and creates resources
+func DefaultHelmClient() HelmClient {
+	return &defaultHelmClient{
+		fs:              afero.NewOsFs(),
+		tempFile:        NewTempFile(),
+		resourceFetcher: NewDefaultResourceFetcher(),
+		helmLoaders:     NewHelmLoaders(),
+	}
+}
+
+type defaultHelmClient struct {
+	fs              afero.Fs
+	tempFile        TempFile
+	resourceFetcher ResourceFetcher
+	helmLoaders     HelmLoaders
+}
+
+func NewDefaultHelmClient(
+	fs afero.Fs,
+	tempFile TempFile,
+	resourceFetcher ResourceFetcher,
+	helmLoaders HelmLoaders) *defaultHelmClient {
+	return &defaultHelmClient{
+		fs:              fs,
+		tempFile:        tempFile,
+		resourceFetcher: resourceFetcher,
+		helmLoaders:     helmLoaders,
+	}
+}
+
+func (d *defaultHelmClient) NewInstall(namespace, releaseName string, dryRun bool) (HelmInstall, *cli.EnvSettings, error) {
+	actionConfig, settings, err := d.helmLoaders.ActionConfigLoader.NewActionConfig(namespace)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	client := action.NewInstall(actionConfig)
+	client.ReleaseName = releaseName
+	client.Namespace = namespace
+	client.DryRun = dryRun
+
+	// If this is a dry run, we don't want to query the API server.
+	// In the future we can make this configurable to emulate the `helm template --validate` behavior.
+	client.ClientOnly = dryRun
+
+	return client, settings, nil
+}
+
+func (d *defaultHelmClient) NewUninstall(namespace string) (HelmUninstall, error) {
+	actionConfig, _, err := d.helmLoaders.ActionConfigLoader.NewActionConfig(namespace)
+	if err != nil {
+		return nil, err
+	}
+	return action.NewUninstall(actionConfig), nil
+}
+
+func (d *defaultHelmClient) DownloadChart(chartArchiveUri string) (*chart.Chart, error) {
+
+	// 1. Get a reader to the chart file (remote URL or local file path)
+	chartFileReader, err := d.resourceFetcher.GetResource(chartArchiveUri)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = chartFileReader.Close() }()
```

Comment:
```suggestion
	defer func() { chartFileReader.Close() }()
```

Reply:
You don't prefer explicitly ignoring return values?